utils/kafka-test-utils/src/main/scala/pl/touk/nussknacker/engine/kafka/KafkaClient.scala

```diff
@@ -1,56 +1,136 @@
 
   private lazy val adminClient = KafkaUtils.createKafkaAdminClient(KafkaConfig(Some(Map("bootstrap.servers" -> kafkaAddress)), None))
 
-  def createTopic(name: String, partitions: Int = 5): Unit = {
+  def createTopic(name: String, partitions: Int = 5): Unit =
     adminClient.createTopics(Collections.singletonList(new NewTopic(name, partitions, 1: Short))).all().get()
-  }
 
-  def deleteTopic(name: String): Unit = {
+  def deleteTopic(name: String): Unit =
     adminClient.deleteTopics(util.Arrays.asList(name)).all().get()
-  }
 
-  def topic(name: String): Option[TopicDescription] = Try(adminClient.describeTopics(util.Arrays.asList(name)).allTopicNames().get()).toOption.map(_.get(name))
+  def topic(name: String): Option[TopicDescription] =
+    Try(adminClient.describeTopics(util.Arrays.asList(name)).allTopicNames().get()).toOption.map(_.get(name))
+
+  def sendRawMessage(topic: String, content: Array[Byte]): Future[RecordMetadata] =
+    sendRawMessage(topic, null, content)
 
   def sendRawMessage(topic: String, key: Array[Byte], content: Array[Byte], partition: Option[Int] = None, timestamp: java.lang.Long = null, headers: Headers = KafkaRecordUtils.emptyHeaders): Future[RecordMetadata] = {
     val promise = Promise[RecordMetadata]()
-    val record = partition.map(new ProducerRecord[Array[Byte], Array[Byte]](topic, _, timestamp, key, content, headers))
-      .getOrElse(new ProducerRecord[Array[Byte], Array[Byte]](topic, null, timestamp, key, content, headers))
+    val record = createRecord(topic, key, content, partition, timestamp, headers)
     rawProducer.send(record, producerCallback(promise))
     promise.future
   }
 
-  def sendMessage(topic: String, key: String, content: String, partition: Option[Int] = None): Future[RecordMetadata] = {
+  def sendMessage[T: Encoder](topic: String, content: T): Future[RecordMetadata] =
+    sendMessage(topic, null, content)
+
+  def sendMessage[T: Encoder](topic: String, key: String, content: T, partition: Option[Int] = None, timestamp: java.lang.Long = null, headers: Headers = KafkaRecordUtils.emptyHeaders): Future[RecordMetadata] = {
+    val strContent = content match {
+      case str: String => str
+      case _ => implicitly[Encoder[T]].apply(content).noSpaces
+    }
+
     val promise = Promise[RecordMetadata]()
-    val record = partition.map(new ProducerRecord[String, String](topic, _, key, content)).getOrElse(new ProducerRecord[String, String](topic, key, content))
+    val record = createRecord(topic, key, strContent, partition, timestamp, headers)
     producer.send(record, producerCallback(promise))
     promise.future
   }
 
-  def sendMessage(topic: String, content: String): Future[RecordMetadata] = {
-    val promise = Promise[RecordMetadata]()
-    producer.send(new ProducerRecord[String, String](topic, content), producerCallback(promise))
-    promise.future
+  def consumeLastMessage[T: Decoder](topic: String): KeyMessage[String, T] =
```

Comment:
Is it really more readable now? Now someone new has to think about which one from all these variants of methods to pick :-/ Having so many methods is kind of anti-pattern. Because of that people designed builders

Reply:
Ok, you're right -> I'm gonna remove all *LastMessage.

Reply:
I've also remove consumeStrMessages, right now we have just consumeMessages and consumeRawMessages